src/test/java/org/opensearch/knn/quantization/quantizationState/QuantizationStateTests.java

```diff
@@ -65,4 +66,30 @@ public void testSerializationWithDifferentVersions() throws IOException {
         assertArrayEquals(mean, deserializedState.getMeanThresholds(), delta);
         assertEquals(params.getSqType(), deserializedState.getQuantizationParams().getSqType());
     }
+
+    public void testOneBitScalarQuantizationStateRamBytesUsed() {
+        ScalarQuantizationParams params = new ScalarQuantizationParams(ScalarQuantizationType.ONE_BIT);
+        float[] mean = { 1.0f, 2.0f, 3.0f };
+
+        OneBitScalarQuantizationState state = new OneBitScalarQuantizationState(params, mean);
+        long expectedRamBytesUsed = RamUsageEstimator.shallowSizeOfInstance(OneBitScalarQuantizationState.class) + RamUsageEstimator
+            .shallowSizeOf(params) + RamUsageEstimator.sizeOf(mean);
+
+        assertEquals(expectedRamBytesUsed, state.ramBytesUsed());
+    }
+
+    public void testMultiBitScalarQuantizationStateRamBytesUsed() {
+        ScalarQuantizationParams params = new ScalarQuantizationParams(ScalarQuantizationType.TWO_BIT);
+        float[][] thresholds = { { 0.5f, 1.5f, 2.5f }, { 1.0f, 2.0f, 3.0f } };
+
+        MultiBitScalarQuantizationState state = new MultiBitScalarQuantizationState(params, thresholds);
+        long expectedRamBytesUsed = RamUsageEstimator.shallowSizeOfInstance(MultiBitScalarQuantizationState.class) + RamUsageEstimator
```

Comment:
Could we test based on a formula as opposed to RamUsageEstimator? I just want to ensure that these are giving us what we are expecting. If not, could you paste on this comment via log what the estimate is for the tests and see if we can ensure theyre about what we expect?